internal/db/db.go

```diff
@@ -0,0 +1,44 @@
+package db
+
+import (
+	"os"
+	"path/filepath"
+
+	"go.etcd.io/bbolt"
+)
+
+var Database *bbolt.DB
+
+const PinnedBucket = "PinnedNotes"
+
+// Initialize the database and ensure the bucket exists
+func InitializeDB() error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	db_path := filepath.Join(homeDir, ".config", "oolong", "oolong.db")
```

Comment:
Also I think the conventional place for the db file would actually be in `~/.local/share/oolong` since it isn't really a configuration file